components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/internal/handler/CORSRequestHandler.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ * NOTE: The code/logic in this class is copied from https://bitbucket.org/thetransactioncompany/cors-filter.
+ * All credits goes to the original authors of the project https://bitbucket.org/thetransactioncompany/cors-filter.
+ */
+
+package org.wso2.carbon.identity.cors.valve.internal.handler;
+
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceClientException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceServerException;
+import org.wso2.carbon.identity.cors.mgt.core.model.CORSConfiguration;
+import org.wso2.carbon.identity.cors.mgt.core.model.Origin;
+import org.wso2.carbon.identity.cors.mgt.core.model.ValidatedOrigin;
+import org.wso2.carbon.identity.cors.service.CORSManager;
+import org.wso2.carbon.identity.cors.valve.constant.ErrorMessages;
+import org.wso2.carbon.identity.cors.valve.constant.Header;
+import org.wso2.carbon.identity.cors.valve.exception.CORSException;
+import org.wso2.carbon.identity.cors.valve.internal.CORSValveServiceHolder;
+import org.wso2.carbon.identity.cors.valve.internal.util.HeaderUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Request handler for the CORS valve.
+ */
+public class CORSRequestHandler {
+
+    /**
+     * Get CORSManager instance.
+     *
+     * @return CORSManager
+     */
+    private CORSManager getCORSManager() {
+
+        return CORSValveServiceHolder.getInstance().getCorsManager();
+    }
+
+    /**
+     * Handles a simple or actual CORS request.
+     *
+     * @param request  The HTTP request.
+     * @param response The HTTP response.
+     * @throws CORSException                  If the request is invalid or denied.
```

Comment:
formatting issue

Reply:
I think it is the correct formatting as the description in the error should be indented after the length of the other errors, which is `CORSManagementServiceException`